astropy/units/physical.py

```diff
@@ -476,7 +476,7 @@ def def_physical_type(unit, name):
         _attrname_physical_mapping[attr_name] = physical_type
 
 
-def get_physical_type(obj):
+def get_physical_type(obj: object) -> PhysicalType:
```

Comment:
Seems good to have `QuantityLike` and `PhysicalTypeLike` defined, use it here.

Reply:
:+1: on using a type alias `PhysicalTypeLike`. This is again driven by use of type annotations as developer docs, not trying to please mypy. Whether `PhysicalTypeLike` tries to enumerate all the possibilities or if it is just an empty type alias makes no difference to me, but seeing that while developing code is very useful.

Reply:
Indeed, and I like that those can be stored in a `types.py` file, so one can fiddle with those definitions without having to change how the code itself looks.